### DISCUSSION TITLE
NO_PROXY env variable setting

### DISCUSSION THREAD

**Discussion by bbusioc** [minio/minio#14523] [Q&A]
Hello,

I'm trying to install and configure minio in a K8S cluster using the helm charts.
For authentication, I'm trying to connect minio to a keycloak instance which is accessible only if I set a http_proxy environment variable.
Right after setting this variable, I can see in minio pods logs these type of error:

<pre>
Error: Marking http://minio-0.minio-svc.[my-namespace].svc.cluster.local:9000/minio/storage/export/v43 temporary offline; caused by Post "http://minio-0.minio-svc.[my-namespace].svc.cluster.local:9000/minio/storage/export/v43/readall?disk-id=&file-path=format.json&volume=.minio.sys": lookup minio-0.minio-svc.[my-namespace].svc.cluster.local on 192.168.0.10:53: no such host (*fmt.wrapError)
</pre>

I tried setting also the NO_PROXY environment variable in many ways:

- explicit setting of each URL (I'm testing with just 4 replicas, so it was not very complicated)
- adding .minio-svc.[my-namespace].svc.cluster.local
- adding .[my-namespace].svc.cluster.local
- adding .svc.cluster.local
- etc.

Did anyone faced similar issue so far, any suggestions what else I could try?

Thanks,
Bogdan

**Comment by harshavardhana** [comment]
You cannot use proxy settings on MinIO pods @bbusioc we need direct access to your keycloack instance server cannot be expected to go through proxy for the time being.

**Reply by bbusioc** [reply]
ok, thanks for the answer.
kind of a blocking point for me...
do have any idea if there is a plan to add such a configuration in the near future?

**Reply by bbusioc** [reply]
I could connect minio to a keycloak started in the same namespace and use a service name in the MINIO_IDENTITY_OPENID_CONFIG_URL instead of a proxy reachable URL.
After integrate this keycloak with the target OIDC.
But I still have an issue...
In this case the "SSO Login URL" in the console is pointing to the URL used in MINIO_IDENTITY_OPENID_CONFIG_URL, in this case the service name which obviously is not reachable from my browser.
Is there any config that might overwrite this URL in the console homepage?

Thanks,
Bogdan

**Reply by harshavardhana** [reply]
> I could connect minio to a keycloak started in the same namespace and use a service name in the MINIO_IDENTITY_OPENID_CONFIG_URL instead of a proxy reachable URL. After integrate this keycloak with the target OIDC. But I still have an issue... In this case the "SSO Login URL" in the console is pointing to the URL used in MINIO_IDENTITY_OPENID_CONFIG_URL, in this case the service name which obviously is not reachable from my browser. Is there any config that might overwrite this URL in the console homepage?
> 
> Thanks, Bogdan

Set storage-configuration secret in tenant namespace

**Reply by bbusioc** [reply]
sorry, but I'm a bit confused. could you please be more explicit?
